### DISCUSSION TITLE
1.4.11 WCAG on toggle button

### DISCUSSION THREAD

**Discussion by AccessibilityEnthusiastic** [w3c/wcag#3227] [Q&A] [answered]
Hi Team,

I have a query on 1.4.11 WCAG on the selected toggle button.

Here, Are we supposed to test all the below-mentioned conditions?

1. The toggle button's internal background(light blue) with external white background needs to have a 3:1 contrast ratio.
2. The toggle button's internal background(light blue) with Round toggle(blue) needs to have a 3:1 contrast ratio.
3. Round toggle(blue) with external white background needs to have 3:1 contrast ratio..

Please suggest.

![image](https://github.com/w3c/wcag/assets/115781693/5d15c801-9ef9-4eb6-82b4-c138ace4062a)

**Comment by JAWS-test** [comment]
The question is always: Can I see the element type (toggle button) and status (on or off) when I can't see contrasts below 3:1. Therefore the following answers:
*    The toggle button's internal background(light blue) with external white background needs to have a 3:1 contrast ratio. --> YES
*    The toggle button's internal background(light blue) with Round toggle(blue) needs to have a 3:1 contrast ratio. --> NO
*    Round toggle(blue) with external white background needs to have 3:1 contrast ratio. --> YES

**Reply by AccessibilityEnthusiastic** [reply]
@JAWS-test   @patrickhlauke   In this case, does the slider still need to meet the 3:1 contrast requirement, considering that an alternative interaction is provided through numbered text?

<img width="593" height="84" alt="image" src="https://github.com/user-attachments/assets/052f002f-49df-4bc5-8b42-3308746263fc" />

**Reply by bruce-usab** [reply, marked as answer]
@AccessibilityEnthusiastic , I cannot tell what is going on with your image, but that might be an artifact of your image.
- The green square slider (?) needs to have 3:1 contrast against its background.
- The grey-on-while text does not have enough contrast.
- In WCAG parlance, "an alternative interaction" might be your conforming alternative version, and can be legitimate, but auditors are understandably skeptical about such assertions.
